doc/lua_api.txt

```diff
@@ -234,6 +235,40 @@ stripping out the file extension:
 * e.g. `foomod_foothing.png`
 * e.g. `foomod_foothing`
 
+Texture pack directory structure
+--------------------------------
+
+    textures
+    |-- Texture Pack
+    |   |-- screenshot.png
+    |   |-- description.txt
+    |   |-- your_texture_1.png
+    |   |-- your_texture_2.png
+    `-- Another Texture Pack
+
+### Texture Pack
+This is a directory containing the entire contents of a single texture pack.
+It can be chosen more or less freely and will also become the name of the
+texture pack. The name must not be “base”.
+
+### `description.txt`
+A file containing a short description of the texture pack to be shown in the
+texture packs tab.
+
+### `screenshot.png`
+A preview image showing an in-game screenshot of this texture pack; it will be
+shown in the texture packs tab. It should have an aspect ratio of 3:2 and a
+minimum size of 300×200 pixels.
+
+### `your_texture_1.png`, `your_texture_2.png`, etc.
+Any other PNG files will be interpreted as textures. They must have the same
+names as the textures they are supposed to override. For example, to override
+the apple texture of Minetest Game, add a PNG file named `default_apple.png`.
```

Comment:
@Wuzzy2 it doesn't need to be a PNG file, it can also be another image file which is supported by irrlicht, e.g. jpg, it is possible to give a jpg file a wrong ending: ".png"